marklogic-unit-test-client/src/main/java/com/marklogic/test/unit/JaxpServiceResponseUnmarshaller.java

```diff
@@ -67,23 +67,32 @@ public TestSuiteResult parseTestSuiteResult(String xml) {
 		for (int i = 0; i < tests.getLength(); i++) {
 			Element testNode = (Element) tests.item(i);
 			String testName = testNode.getAttribute("name");
+
 			double testTime = Double.parseDouble(testNode.getAttribute("time"));
 			NodeList resultNodes = testNode.getChildNodes();
+      int testTotal = 0;
+      List<Element> successNodes = new ArrayList<>();
+      List<Element> failedNodes = new ArrayList<>();
+      for (int j = 0; j < resultNodes.getLength(); j++) {
+        // An XQuery test module may return additional nodes, such as text nodes, which should be
+        // ignored unless they are element nodes.
+        if (resultNodes.item(j).getNodeType() == ELEMENT_TYPE) {
+          Element resultNode = (Element) resultNodes.item(j);
+          if (resultNode.getTagName().equalsIgnoreCase("test:result")) {
```

Comment:
It seems safer to do this:

    if ("result".equals(resultNode.getLocalName())

That way, we're not assuming that a particular namespace prefix is being used. And the only issue would be if there's somehow a "result" element in a different namespace, which seems very unlikely - though we could check that as well.